buf/registry/module/v1/label_service.proto

```diff
@@ -167,6 +167,9 @@ message ListLabelHistoryRequest {
     (buf.validate.field).string.uuid = true,
     (buf.validate.field).ignore = IGNORE_IF_UNPOPULATED
   ];
+  // Whether or not to only return Commits where the content changed
+  // compared to the previous Commit in the history of this Label.
+  bool only_content_changes = 7;
```

Comment:
This feels like a UI-only need that can be deduced from the information present in the API already - unless there's a strong reason otherwise (maybe there is), we purposefully designed registry-protoc to get out of the practice of making lots of API compromises in the name of a single call pattern

Reply:
The current API makes it possible to get this information, but not in a performant way in the general case. For repositories with a lot of commits that don't actually change the modules in them (i.e. monorepos with a lot of non-protobuf commits), clients will have to page through an arbitrary number of requests of this endpoint to find/fill one page of actual results to display to a user if the user only cares to see commits that only actually changed a module. This is why a filter flag here seems necessary.

We could drop this from the BSR design, but it seems like an important and valuable use case in general (similar to `git log -- [filename]`). I also think it would be a valuable flag to have on the CLI too when fetching the history for a label (but we don't need to do this now). Given all that, I don't really see this as a compromise in the API, we are just making it possible to satisfy a valid/valuable use case.

Reply:
Ok - I think I would name this field "skip_no_diff_commits", or something to do with "skip" and "commits" (open to other suggestions)